google/resource_sql_database_instance.go

```diff
@@ -1062,7 +964,21 @@ func flattenAuthorizedNetworks(entries []*sqladmin.AclEntry) interface{} {
 		networks = append(networks, data)
 	}
 
-	return networks
+	return schema.NewSet(schema.HashResource(sqlDatabaseAuthorizedNetWorkSchemaElem), networks)
+}
+
+func expandAuthorizedNetworks(configured *schema.Set) []*sqladmin.AclEntry {
+	entries := make([]*sqladmin.AclEntry, 0)
+	for _, authNetworkConfig := range configured.List() {
+		entry := authNetworkConfig.(map[string]interface{})
+
+		entries = append(entries, &sqladmin.AclEntry{
```

Comment:
If this resource were ForceNew I think this (putting everything in the struct instead of checking if it's set first) would work, but since it's updatable you'll need `ForceSendFields` in order to be able to change values from non-empty to empty. But then by supplying `ForceSendFields`, you're insisting that each value has a default of empty, and I haven't checked whether that's actually true. If it is, can you add a comment as to why this is ok? (likewise with `expandIpConfiguration`)

Reply:
Ok the actual more important comment I meant to make here: If any of these aren't set this will panic because you can't type-assert nil to a string.

Reply:
Actually, check me on that. I'm not sure whether it'll be nil or the zero value.